reporting/api/src/main/java/org/jboss/windup/reporting/config/HasHint.java

```diff
@@ -34,51 +35,31 @@ public boolean evaluate(GraphRewrite event, EvaluationContext context, WindupVer
         ExecutionStatistics.get().begin(HasHint.class.getCanonicalName());
         try
         {
-            boolean result = false;
             InlineHintService service = new InlineHintService(event.getGraphContext());
 
-            if (payload instanceof FileReferenceModel)
-            {
-                Iterable<InlineHintModel> hints = service.getHintsForFileReference((FileReferenceModel) payload);
-                if (messagePattern == null)
-                {
-                    result = hints.iterator().hasNext();
-                }
-                else
-                {
-                    for (InlineHintModel c : hints)
-                    {
-                        ParameterizedPatternResult parseResult = messagePattern.parse(c.getHint());
-                        if (parseResult.matches() && parseResult.isValid(event, context))
-                        {
-                            result = true;
-                            break;
-                        }
-                    }
-                }
-            }
+            Iterable<InlineHintModel> hints = null;
+            // FileLocationModel is a fake FileReferenceModel with [1,1] row and column, so we want hints for the whole file.
```

Comment:
Yes, I've made a rushed conclusion. I think this will need a change in how `File` creates FileLocationModel.